### PR TITLE
PHP8 errors and several other fixes

### PR DESCRIPTION
**Pull request by brainfoolong**
I'm testing PHP8 with our apps that use TCPDF. There are things in TCPDF that are incompatible with PHP8 and throw errors. I've took the time to fix the issues and also fixed other issues that i came across.

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1 @@
+*.php text eol=lf
```
Comment (williamdes) [inline .gitattributes:1]: Could you export ignore `.github` ?

File: include/tcpdf_static.php [hunk 2]
```diff
@@ -1859,7 +1859,7 @@ public static function url_exists($url) {
 	public static function encodeUrlQuery($url) {
 		$urlData = parse_url($url);
 		if (isset($urlData['query']) && $urlData['query']) {
-			$urlQueryData = array();
+			$urlQueryData = [];
```
Comment (williamdes) [inline include/tcpdf_static.php:1862]: You should revert this change to keep maximal php versions coverage

File: .gitattributes (hunk 1 above)
Comment (brainfoolong) [reply .gitattributes:1]: Hi, what do mean with "export"?

File: tcpdf.php [hunk 3]
```diff
@@ -4291,6 +4291,7 @@ public function AddFont($family, $style='', $fontfile='', $subset='default') {
 				$fontfile = TCPDF_FONTS::getFontFullPath($tmp_fontfile, $fontdir);
 			}
 		}
+		$originalsize = $size1 = $size2 = null;
```
Comment (williamdes) [inline tcpdf.php:4294]:
```suggestion
		$originalsize = null;
		$size1 = null;
		$size2 = null;
```

File: include/tcpdf_static.php (hunk 2 above)
Comment (brainfoolong) [reply include/tcpdf_static.php:1862]: I saw that in several other parts of the code that are already short array syntax is used, so i've changed it there too.

Comment (brainfoolong) [reply include/tcpdf_static.php:1862]: Ah, nevermind. I was wrong, i'll revert.

File: .gitattributes (hunk 1 above)
Comment (williamdes) [reply .gitattributes:1]: Hi
The file controls `git archive` and basically, excluding some files nobody wants on their production servers is always cool. In debian we do not need the github folder for example 

https://github.com/williamdes/mariadb-mysql-kbs/blob/main/.gitattributes#L15
https://salsa.debian.org/phpmyadmin-team/mariadb-mysql-kbs/-/tree/upstream

Tcpdf: https://salsa.debian.org/phpmyadmin-team/tcpdf/-/tree/upstream

Comment (brainfoolong) [reply .gitattributes]: Thx. I've done that.

Comment (williamdes) [reply .gitattributes]: Thank you!
One note about your commit message it is more "do not export github folder in vendor tarballs" ;)

Comment (brainfoolong) [reply .gitattributes]: Thanks for the info.

**Comment by mircobabini** [comment]
I know that you want to keep doing the great job your are doing but... when will you push this hotfix? It's causing us a lot of issues. :) Thanks.

**Comment by williamdes** [comment]
@nicolaasuni is this PR rejected or merged in some way?

**Comment by williamdes** [comment]
@nicolaasuni Was the PR closed by mistake ?

**Comment by nicolaasuni** [comment]
This was closed because the main branch is now called "main", so please resend to the correct branch.
